snuba/clickhouse/optimize/optimize.py

```diff
@@ -123,9 +127,19 @@ def run_optimize_cron_job(
     except NoOptimizedStateException:
         # We don't have any recorded state of partitions needing optimization
         # for today. So we need to build it.
-        partitions = get_partitions_to_optimize(
+        partitions = get_partitions_from_clickhouse(
             clickhouse, storage, database, table, before
         )
+
+        if divide_partitions_count > 0:
```

Comment:
Nit: This will always be true since the default value is 1. Probably skip this check or change it to `> 1`